component/module/module.go

```diff
@@ -33,13 +33,13 @@ type Exports struct {
 }
 
 // NewModuleComponent initializes a new ModuleComponent.
-func NewModuleComponent(o component.Options) ModuleComponent {
+func NewModuleComponent(o component.Options) *ModuleComponent {
```

Comment:
Not necessary for this PR but I think a pointer may be more correct for a new function return